### PR TITLE
Increase gossip size on first push

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

This improves E2E TTF of issued transactions.

## How this works

Increases the number of validators that unsent transactions are gossiped to.

## How this was tested

- [X] Updated unit tests

### CODE REVIEW

File: network/p2p/gossip/gossip.go [hunk 1]
```diff
@@ -348,46 +364,48 @@ func (p *PushGossiper[T]) Gossip(ctx context.Context) error {
...
+		p.discarded, // Mark dropped sent transactions as discarded
+	)
+	if err != nil {
+		return fmt.Errorf("unexpected error during regossip: %w", err)
 	}
+	return nil
+}
 
-	maxLastGossipTimeToRegossip := now.Add(-p.maxRegossipFrequency)
+func (p *PushGossiper[T]) gossip(
```
Comment (patrick-ogrady) [inline network/p2p/gossip/gossip.go:393]: ❤️

File: vms/avm/network/config.go [hunk 2]
```diff
@@ -34,12 +36,18 @@ type Config struct {
...
 	TargetGossipSize int `json:"target-gossip-size"`
-	// PushGossipNumValidators is the number of validators to push transactions
-	// to per round of gossip.
-	PushGossipNumValidators int `json:"push-gossip-num-validators"`
-	// PushGossipNumPeers is the number of peers to push transactions to per
-	// round of gossip.
-	PushGossipNumPeers int `json:"push-gossip-num-peers"`
+	// PushGossipFirstNumValidators is the number of validators to push
+	// transactions to in the first round of gossip.
+	PushGossipFirstNumValidators int `json:"push-gossip-first-num-validators"`
```
Comment (patrick-ogrady) [inline vms/avm/network/config.go:41]: nit: `initial` seems clearer to me

**Review by patrick-ogrady — APPROVED** [review]

**Review by dhrubabasu — APPROVED** [review]

**Review by abi87 — APPROVED** [review]

File: network/p2p/gossip/gossip.go [hunk 3]
```diff
@@ -348,46 +364,48 @@ func (p *PushGossiper[T]) Gossip(ctx context.Context) error {
...
+		ctx,
+		now,
+		p.gossipParams,
+		p.toGossip,
+		p.toRegossip,
+		&cache.Empty[ids.ID, struct{}]{}, // Don't mark dropped unsent transactions as discarded
 	)
+	if err != nil {
+		return fmt.Errorf("unexpected error during gossip: %w", err)
+	}
```
Comment (joshua-kim) [inline network/p2p/gossip/gossip.go:377]:
```suggestion
	if err := p.gossip(
		ctx,
		now,
		p.gossipParams,
		p.toGossip,
		p.toRegossip,
		&cache.Empty[ids.ID, struct{}]{}, // Don't mark dropped unsent transactions as discarded
	); err != nil {
		return fmt.Errorf("unexpected error during gossip: %w", err)
	}
```

File: network/p2p/gossip/gossip.go [hunk 4]
```diff
@@ -348,46 +364,48 @@ func (p *PushGossiper[T]) Gossip(ctx context.Context) error {
...
+		ctx,
+		now,
+		p.regossipParams,
+		p.toRegossip,
+		p.toRegossip,
+		p.discarded, // Mark dropped sent transactions as discarded
+	)
+	if err != nil {
+		return fmt.Errorf("unexpected error during regossip: %w", err)
 	}
```
Comment (joshua-kim) [inline network/p2p/gossip/gossip.go:389]:
```suggestion
	if err := p.gossip(
		ctx,
		now,
		p.regossipParams,
		p.toRegossip,
		p.toRegossip,
		p.discarded, // Mark dropped sent transactions as discarded
	); err != nil {
		return fmt.Errorf("unexpected error during regossip: %w", err)
	}
```

**Pull request merged**